services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java

```diff
@@ -264,36 +265,67 @@ public Response getProjectsStatistics(@ApiParam(value = "id", required = true) @
     try {
-    ListAccess<User> usersList = userService.findUserByName(query);
-    JSONArray usersJsonArray = new JSONArray();
-    for (User user : usersList.load(0, UserUtil.SEARCH_LIMIT)) {
+      ProjectDto project = projectService.getProject(projectId);
+      Set<String> projectParticipantList = project.getParticipator();
+      JSONArray usersJsonArray = new JSONArray();
+      HashSet<org.exoplatform.social.core.identity.model.Identity> userIdentities = new HashSet<>();
+      projectParticipantList.forEach(participant -> {
+        if (!participant.startsWith(SPACE_PREFIX)) {
+          if (!StringUtils.equals(currentUser.getUserId(), participant) && participant.contains(query)) {
+            org.exoplatform.social.core.identity.model.Identity identity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, participant);
+            userIdentities.add(identity);
+          }
+        } else {
+          Space space = spaceService.getSpaceByPrettyName(participant.substring(SPACE_PREFIX.length()));
+          if (space != null) {
+            searchSpaceMembers(userIdentities, space, query, currentUser);
+          }
+        }
+      });
+      userIdentities.forEach(identity -> {
+        addParticipantToUserList(usersJsonArray, (org.exoplatform.social.core.identity.model.Identity) identity);
+      });
+      return Response.ok(usersJsonArray.toString()).build();
+    } catch (Exception e) {
+      LOG.error("Can't get Users ", e);
+      return Response.serverError().entity(e.getMessage()).build();
+    }
+  }
+
+  private void addParticipantToUserList(JSONArray usersJsonArray, org.exoplatform.social.core.identity.model.Identity identity) {
+    if (identity.isEnable() && !identity.isDeleted()) {
       JSONObject userJson = new JSONObject();
-      Space space = spaceService.getSpaceByPrettyName(projectName);
-      if (space == null || spaceService.isMember(space, user.getUsername())) {
-        userJson.put("username", user.getUsername());
-        userJson.put("fullname", user.getDisplayName());
-        userJson.put("avatar", user.getAvatar());
+      try {
+        userJson.put("id", identity.getRemoteId());
+        userJson.put("name", identity.getProfile().getFullName());
+        userJson.put("avatar", identity.getProfile().getAvatarUrl());
+        userJson.put("type", "contact");
         usersJsonArray.put(userJson);
+      } catch (JSONException e) {
+        throw new IllegalStateException("Error while adding participant to JSONArray", e);
       }
     }
-    return Response.ok(usersJsonArray.toString()).build();
-  }
-    catch (Exception e) {
-            LOG.error("Can't get Users ",e);
-            return Response.serverError().entity(e.getMessage()).build();
-            }
   }
 
+  private void searchSpaceMembers(HashSet<org.exoplatform.social.core.identity.model.Identity> userIdentities, Space space, String query, Identity currentUser) {
+    String[] spaceMembers = space.getMembers();
```

Comment:
Could you use a searching API instead of loading all spac emembers, this may cause performance issues with populated spaces.

Reply:
Already exist in `getProjectParticipants()`